src/components/Search/SearchRouter/SearchRouter.tsx

```diff
@@ -175,106 +183,119 @@ function SearchRouter({onRouterClose}: SearchRouterProps) {
                     const filteredParticipants = participantsAutocompleteList
-                        .filter((participant) => participant.includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(participant.toLowerCase()))
+                        .filter((participant) => participant.name.includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(participant.name.toLowerCase()))
                         .sort()
                         .slice(0, 10);
-                    setAutocompleteSuggestions(filteredParticipants.map((participant) => ({text: `${CONST.SEARCH.SYNTAX_FILTER_KEYS.FROM}:${participant}`, query: `${participant}`})));
-                    return;
+                    filteredAutocompleteSuggestions = filteredParticipants.map((participant) => ({
+                        filterKey: CONST.SEARCH.SYNTAX_FILTER_KEYS.FROM,
+                        text: participant.name,
+                        autocompleteID: participant.accountID,
+                    }));
+                    break;
                 }
                 case CONST.SEARCH.SYNTAX_FILTER_KEYS.TO: {
                     const filteredParticipants = participantsAutocompleteList
-                        .filter((participant) => participant.includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(participant.toLowerCase()))
+                        .filter((participant) => participant.name.includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(participant.name.toLowerCase()))
                         .sort()
                         .slice(0, 10);
-                    setAutocompleteSuggestions(filteredParticipants.map((participant) => ({text: `${CONST.SEARCH.SYNTAX_FILTER_KEYS.TO}:${participant}`, query: `${participant}`})));
-                    return;
+                    filteredAutocompleteSuggestions = filteredParticipants.map((participant) => ({
+                        filterKey: CONST.SEARCH.SYNTAX_FILTER_KEYS.TO,
+                        text: participant.name,
+                        autocompleteID: participant.accountID,
+                    }));
+                    break;
                 }
                 case CONST.SEARCH.SYNTAX_FILTER_KEYS.IN: {
                     const filteredChats = searchOptions.recentReports
                         .filter((chat) => chat.text?.toLowerCase()?.includes(autocompleteValue.toLowerCase()))
                         .sort((chatA, chatB) => (chatA > chatB ? 1 : -1))
                         .slice(0, 10);
-                    setAutocompleteSuggestions(filteredChats.map((chat) => ({text: `${CONST.SEARCH.SYNTAX_FILTER_KEYS.IN}:${chat.text}`, query: `${chat.reportID}`})));
-                    return;
+                    filteredAutocompleteSuggestions = filteredChats.map((chat) => ({
+                        filterKey: CONST.SEARCH.SYNTAX_FILTER_KEYS.IN,
+                        text: chat.text ?? '',
```

Comment:
I think this is ok for now, but we might have to add some more details to the name to help users differentiate multiple reports with the same name, e.g. `admin` rooms

Reply:
I am following the slack thread for this and will ask there

Reply:
TLDR this will be implemented as a holistic approach throughout the app, so we decided to do nothing for now.